File: Lab3/src/main/java/ParamsGenerator.java
import lombok.AllArgsConstructor;

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
public class ParamsGenerator {
    private int M;

    public  List<Pair<Integer>> findParams() {
        var availableParams = new ArrayList<Pair<Integer>>();
        for(int i = 0; i < M; ++i) {
            for(int j = 0; j < M; ++j) {
                if(BigInteger.valueOf(4L * (int)Math.pow(i, 3) + 27L * (int)Math.pow(j, 2)).mod(BigInteger.valueOf(M)).intValue() != 0) {
                    availableParams.add(new Pair<Integer>(i, j));
                }
            }
        }
        return availableParams;
    }
}

File: Lab3/src/main/java/ParamsGenerator.kt
import java.util.ArrayList
import kotlin.math.pow

class ParamsGenerator(private val M: Int) {
    fun findParams(): List<Pair<Int, Int>> {
        val availableParams = ArrayList<Pair<Int, Int>>()
        for (i in 0 until M) {
            for (j in 0 until M) {
                if ((4 * i.toDouble().pow(3).toInt() + 27 * j.toDouble().pow(2).toInt()) % M != 0) {
                    availableParams.add(Pair(i, j))
                }
            }
        }
        return availableParams
    }
}